libsnark/zk_proof_systems/plonk/utils.hpp

```diff
@@ -227,6 +227,31 @@ void plonk_generate_random_constants_k1_k2(
 template<typename FieldT>
 bool plonk_are_valid_constants_k1_k2(const FieldT &k1, const FieldT &k2);
 
+/// Return a matrix with top N rows filled in with the public inputs
+/// selector vectors where N = num_public_inputs is the number of
+/// public inputs passed as an input parameter by the caller.
+template<typename ppT>
+std::vector<std::vector<libff::Fr<ppT>>> plonk_prepare_gates_matrix(
+    const size_t &num_public_inputs);
+
+/// Extract the values corresponing to the public inputs from the
+/// witness using the respective wire indices passed as input. Those
+/// values are passed on to the verifier together with the proof.
+template<typename ppT>
+std::vector<libff::Fr<ppT>> plonk_public_input_values_from_indices(
+    const std::vector<libff::Fr<ppT>> &witness,
+    const std::vector<size_t> &PI_wire_indices);
+
+/// A wrapper for plonk_public_input_values_from_indices. Extracts the
+/// values corresponing to the public inputs from the witness, assuming
+/// that they are in the first num_public_inputs positions. In other
+/// words extracts the values of the public inputs from the *length* of
+/// the public input vector.
+template<typename ppT>
+std::vector<libff::Fr<ppT>> plonk_public_input_values_from_length(
```

Comment:
I'd suggest simplifying the name here, implying this is the standard approach, with the longer name reserved for the non-standard aproach.
```suggestion
std::vector<libff::Fr<ppT>> plonk_public_input_values(
```

Reply:
Good suggestion. Applied.